lib/rules/no-magic-numbers.js

```diff
@@ -26,10 +26,152 @@ function normalizeIgnoreValue(x) {
 	return x;
 }
 
+/**
+ * Gets the true parent of the literal, handling prefixed numbers (-1 / +1)
+ * @param {ASTNode} node The literal node
+ * @returns {ASTNode} The parent node
+ */
+function getLiteralParent(node) {
+	if (["-", "+"].includes(node.parent.operator)) {
+		return node.parent.parent;
+	}
+
+	return node.parent;
+}
```

Comment:
This looks like a duplicate logic now that `fullNumberNode` includes unary `+`. I think we can remove this function and pass `fullNumberNode` instead of `node` to `isParentTSEnumDeclaration()` and others.

Reply:
Yes, fixed in https://github.com/eslint/eslint/pull/19561/commits/b95eab9c3e16d50ff8f89625d6f6b3cf534fbfd6